GTFO-API/Utilities/Impl/ThreadDispatcher_Impl.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using AssetShards;
+using UnhollowerBaseLib.Attributes;
+using UnityEngine;
+
+namespace GTFO.API.Utilities.Impl
+{
+    internal class ThreadDispatcher_Impl : MonoBehaviour
+    {
+        public ThreadDispatcher_Impl(IntPtr intPtr) : base(intPtr) { }
+
+        static ThreadDispatcher_Impl()
+        {
+            AssetShardManager.add_OnStartupAssetsLoaded((Action)OnAssetsLoaded);
+        }
+
+        [HideFromIl2Cpp]
+        private static void OnAssetsLoaded()
+        {
+            if (s_Instance == null)
+            {
+                GameObject dispatcher = new();
+                var dispatcherComp = dispatcher.AddComponent<ThreadDispatcher_Impl>();
```

Comment:
Explicitly declare the type
```suggestion
                ThreadDispatcher_Impl dispatcherComp = dispatcher.AddComponent<ThreadDispatcher_Impl>();
```